rclcpp/include/rclcpp/generic/create_generic_publisher.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020, Apex.AI Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__GENERIC__CREATE_GENERIC_PUBLISHER_HPP_
+#define RCLCPP__GENERIC__CREATE_GENERIC_PUBLISHER_HPP_
+
+#include <memory>
+#include <string>
+
+#include "rclcpp/generic/generic_publisher.hpp"
+#include "rclcpp/node_interfaces/node_topics_interface.hpp"
+#include "rclcpp/qos.hpp"
+#include "rclcpp/callback_group.hpp"
+
+namespace rclcpp
+{
+namespace generic
+{
```

Comment:
following up on the question why we need a `generic` namespace, I would suggest to integrate this function into the existing `rclcpp/create_publisher.hpp`

Reply:
same for `create_subscription`

Reply:
I can agree with not having another namespace, but I would avoid putting it into the existing files. Just create new ones along side it.

Reply:
Ok, then I will remove the extra directory and the namespace.